Fixes #940 by extending the relative keyword list

DfKimera opened the pull request:
Fixes #940 by extending the relative keyword list to include bare weekdays; added test coverage for this edge case.

u01jmg3 commented on src/Carbon/Carbon.php:
```diff
@@ -98,6 +98,12 @@ class Carbon extends DateTime
+        'tuesday',
+        'thursday',
+        'friday',
+        'saturday',
```
`wednesday`?

DfKimera replied:
Fixed!

kylekatarnls commented:
Hi, I think we could start by merging this but we would need a finest approach, here are some valid relative and absolute times:

```
first day of January 2008 > absolute
first day of next month > relative
sunday noon > relative
sunday nex week > relative
last sunday of January 2008 > absolute
```

And the unit test with `new Carbon` is good but a more atomic test with `hasRelativeKeywords` with more tricky sentences like this would be good too.

Do you want to complete your PR this way? Else I would just merge it and see later for those improvements.

DfKimera commented:
@kylekatarnls I've fixed those edge cases of non-relative dates with relative keywords (now testing for years in the time string too). Also added a bunch of test scenarios for these, and now also testing if the `hasRelativeKeywords` is matching correctly. Should be good for a merge now!

DfKimera commented:
Seems there's an inconsistency in PHP versions regarding the "sunday next week" scenario.
I removed the scenario, as this relates to issues in the underlying `DateTime` and not Carbon.

kylekatarnls commented:
Thanks! Good job.
